### PR TITLE
feat: Add full authentication workflow

### PR DESCRIPTION
**Pull request by lsetiawan**
This PR adds the full authentication logic for login to the app and signing up.

@nikiburggraf apologies for the giant PR, let me know if there's anything that's unclear. I still have yet put more comments throughout the code 😬

### CODE REVIEW

File: src/support_sphere/lib/presentation/pages/auth/login_page.dart [hunk 1]
```diff
@@ -1,139 +1,83 @@
-import 'package:ionicons/ionicons.dart';
+import 'package:flow_builder/flow_builder.dart';
 import 'package:loading_overlay/loading_overlay.dart';
 import 'package:flutter/material.dart';
 import 'package:support_sphere/constants/routes.dart';
```
Comment (lsetiawan) [inline src/support_sphere/lib/presentation/pages/auth/login_page.dart:4]:
```suggestion
```

File: src/support_sphere/lib/presentation/pages/landing_page.dart [hunk 2]
```diff
@@ -1,19 +1,27 @@
+import 'package:flow_builder/flow_builder.dart';
 import 'package:flutter/material.dart';
 import 'package:support_sphere/presentation/components/icon_logo.dart';
 import 'package:support_sphere/constants/string_catalog.dart';
 import 'package:support_sphere/constants/color.dart';
 import 'package:support_sphere/constants/routes.dart';
```
Comment (lsetiawan) [inline src/support_sphere/lib/presentation/pages/landing_page.dart:6]:
```suggestion
```

**Review by nikiburggraf — COMMENTED** [review]
Whew, lot here. Left a small number of comments

A lot of it seems to look good, but it's hard to know for sure without things being hooked up.

Are there any plans to non-manually test the logical parts of this flow? Maybe unit tests with mocking out the database connection? If that's coming later, I feel like there's enough in this PR for it to be okay, but I'd definitely feel more confident about these changes with that level of testing

File: src/support_sphere/lib/data/models/auth_user.dart [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    this.phone,
+  });
+
+  /// The current user's email address.
+  final String? email;
+
+  /// The current user's id.
+  final String uuid;
+
+  /// The current user's name (display name).
```
Comment (lsetiawan) [inline src/support_sphere/lib/data/models/auth_user.dart:24]:
```suggestion
  /// The current user's phone.
```

File: src/support_sphere/lib/data/models/auth_user.dart [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  final String? phone;
```
[the remark this answers is not in the document]
Comment (lsetiawan) [reply src/support_sphere/lib/data/models/auth_user.dart:25]: Great catch! There has been so many changes 😅

File: src/support_sphere/lib/data/repositories/user.dart [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+class UserRepository {
+  AuthUser? _user;
+
+  Future<AuthUser?> getUser() async {
+    if (_user != null) return _user;
+    // TODO: Fetch user from API
+    return Future.delayed(
+      const Duration(milliseconds: 300),
+      () => _user = AuthUser(uuid: const Uuid().v4()),
+    );
```
[the remark this answers is not in the document]
Comment (lsetiawan) [reply src/support_sphere/lib/data/repositories/user.dart:18]: Yea it is.

**Comment by lsetiawan** [comment]
> Are there any plans to non-manually test the logical parts of this flow?

We'll probably add non-manual testing later.. but currently that's definitely not in our plans since there are a lot of features to build out. I definitely hear you on testing the logical parts.. but definitely would have to punt that for now. We'll need to get a running test instance soon so we can at least manually test against it. 😅

**Comment by lsetiawan** [comment]
Following up on my comment above, an API test can definitely done with just logic.. so I might do that for logic code changes. I'll have to follow that up in the future though.

**Pull request merged**